docs/01-search-server.md

```diff
@@ -0,0 +1,82 @@
+# Search engine for BCC
+
+## Status
+
+Proposed
+
+## Context
+
+Currently everyone at BCC is using something custom for the search needs, that works or does not work. For example BCC Media APP uses Algolia, BMM uses the built in capabilities of RavenDB. 
+
+We would like to standardize on one solution that is powerful and flexible enough to be shared amongst the different projects in the organization at large.
+
+### Usage Examples
+
+* Bcc Media search
+	* Full text of content
+	* Title only
+	* ...
+* BMM search
+	* Full text of content
+	* Title
+	* Song numbers
+	* Verses
+* Literature
+	* Full text search
+	* Title Search
+	* Keyword search
+* Portal
+	* Unified search across products
+
+## Requirements
+
+The system must:
+1. be fast and scalable (i.e. we must be able to start small and grow with minimal concern for performance issues when scaled up)
+2. have a billing model we can understand and control
+	* Think that "search as you type" can produce *many* queries
+	* Preferably something that does not scale linearly with number of requests
+1. be a mature product that we have reason to believe will be maintained for the next 3-5 years at least
+2. be able to comply with GDPR (EU hosting at a minimum)
+3. be reasonably easy to use for a developer (no need a math degree to understand what's going on)
```

Comment:
Propose adding some sort of "governance" requirement here. E.g. that access can be granted / revoked ideally based on access groups that are synchronized with our identity directory.